// eslint-plugin-tutorial/lib/rules/es-full-lib.js
/**
 * @fileoverview ...
 * @author 
 */
"use strict";

//------------------------------------------------------------------------------
// Rule Definition
//------------------------------------------------------------------------------

module.exports = {
    meta: {
        docs: {
            description: "...",
            category: "Fill me in",
            recommended: false
        },
        fixable: null,  // or "code" or "whitespace"
        schema: [
            // fill in your schema
        ]
    },

    create: function(context) {
        return {
           CallExpression: function (node) {
                /*returns an error if $ is used instead of jQuery*/
                if(node.callee.name === '$'){
                    context.report({
                        node: node,
                        message: 'Please use jQuery instead of $',
                        fix: function(fixer){
                            /*replaces $ in jQuery*/
                            return fixer.replaceText(node.callee, 'jQuery');
                        }
                    });
                }
            }

        };
    }
};
